### ISSUE TITLE
Ensemble training with Keras

### ISSUE THREAD

**Issue by chivalry123** [fchollet/keras#6434]
Does Keras (plan to) implement Ensemble training, which would possibly be quite practically useful later....

**Comment by StripedBanana** [comment]
I believe it is possible to implement [this](https://arxiv.org/abs/1704.00109) paper with two custom callback and an optimizer function with learning rate decay:

- a custom model checkpointer to save it at the end of every epoch with a different name
- a custom learning rate scheduler to "reboot" the learning rate to its original value

Then just fit your model for say 10 epochs and then, at test time, average the predictions of all models (that's basicall my understanding of what they did in the paper). Note that this usually will "only" get you a few % of accuracy.

**Comment by titu1994** [comment]
I've implemented the Snapshot Ensemble paper in Keras a while back. It should work for Keras 2 without any modifications. 

https://github.com/titu1994/Snapshot-Ensembles

**Comment by StripedBanana** [comment]
Neat! Thanks for sharing.
